Incompatibility with typescript 6/7

https://www.typescriptlang.org/docs/handbook/namespaces-and-modules.html

Apparently in `typescript` 6 and 7 (from `7.0.0-dev.20260222.1` onwards) `dexie@4.3.0` (and probably many previous versions) is not longer valid. We are now getting the following errors:

```
error TS1540: A 'namespace' declaration should not be declared using the 'module' keyword. Please use the 'namespace' keyword instead.
```

[According to a typescript maintainer](https://github.com/microsoft/typescript-go/issues/2873#issuecomment-3942772816), this is silenceable in 6 but from now on will error in 7. From what I can tell, this is probably what `Note: In very old versions of TypeScript namespaces were called ‘Internal Modules’, these pre-date JavaScript module systems.` is referring to, and this might be because the typescript for this was written a long time ago?

In any case, the typescript maintainer is pretty adamant that is something that will need to be fixed in `dexie`. If you agree, I'll submit a patch to simply use `namespace` instead of `module` in the offending places in the codebase.

--- a/src/public/index.d.ts
+++ b/src/public/index.d.ts
@@ -39,7 +39,7 @@ interface _Collection<T,TKey> extends Collection<T,TKey> {}
 
 // Besides being the only exported value, let Dexie also be
 // a namespace for types...
-declare module Dexie {
+declare namespace Dexie {
   // The "Dexie.Promise" type.
   type Promise<T=any> = PromiseExtended<T> // Because many samples have been Dexie.Promise.
   // The "Dexie.Table" interface. Same as named exported interface Table.

